File: client/test/intentClassifier.test.ts
import { describe, it, expect } from 'vitest';
import { classifyIntent } from '../src/utils/protectedText/intentClassifier';

describe('Local intent classifier (Protected Text v2)', () => {
  it('classifies empty string as GENERAL', () => {
    expect(classifyIntent('')).toBe('GENERAL');
  });

  it('classifies urgent messages', () => {
    expect(classifyIntent('This is urgent, please respond now!!')).toBe('URGENT');
    expect(classifyIntent('emergency!!!')).toBe('URGENT');
  });

  it('classifies questions', () => {
    expect(classifyIntent('Are you coming tonight?')).toBe('QUESTION');
    expect(classifyIntent('How are you')).toBe('QUESTION');
  });

  it('classifies time-related messages', () => {
    expect(classifyIntent("Let's meet tomorrow morning")).toBe('TIME');
    expect(classifyIntent('See you at 8pm')).toBe('TIME');
  });

  it('classifies location-related messages', () => {
    expect(classifyIntent('Meet me at the station')).toBe('LOCATION');
  });

  it('classifies requests', () => {
    expect(classifyIntent('Please send the file')).toBe('REQUEST');
    expect(classifyIntent('Could you help with this')).toBe('QUESTION');
  });

  it('classifies negation', () => {
    expect(classifyIntent("No, I don't think so")).toBe('NEGATION');
  });

  it('classifies affirmation', () => {
    expect(classifyIntent('Yes, sounds good')).toBe('AFFIRMATION');
  });

  it('classifies greetings', () => {
    expect(classifyIntent('Hello there')).toBe('GREETING');
    expect(classifyIntent('Good morning!')).toBe('GREETING');
  });

  it('classifies farewells', () => {
    expect(classifyIntent('Bye, take care')).toBe('FAREWELL');
  });

  it('classifies acknowledgements', () => {
    expect(classifyIntent('Got it, thanks')).toBe('ACKNOWLEDGEMENT');
  });

  it('falls back to GENERAL for unclassifiable content', () => {
    expect(classifyIntent('The quarterly report numbers look fine')).toBe('GENERAL');
  });

  it('is deterministic', () => {
    const input = 'Meet me at the station tomorrow at 8pm, urgent!!';
    expect(classifyIntent(input)).toBe(classifyIntent(input));
  });

  it('never throws on unusual input', () => {
    expect(() => classifyIntent('😊'.repeat(50))).not.toThrow();
    expect(() => classifyIntent('你好，世界！')).not.toThrow();
  });
});

File: android/app/src/test/java/com/enctxt/IntentClassifierTest.kt
package com.enctxt

import com.enctxt.core.privacy.IntentCategory
import com.enctxt.core.privacy.IntentClassifier
import org.junit.Assert.*
import org.junit.Test

class IntentClassifierTest {

    @Test
    fun testEmptyStringIsGeneral() {
        assertEquals(IntentCategory.GENERAL, IntentClassifier.classify(""))
    }

    @Test
    fun testClassifiesUrgent() {
        assertEquals(IntentCategory.URGENT, IntentClassifier.classify("This is urgent, please respond now!!"))
        assertEquals(IntentCategory.URGENT, IntentClassifier.classify("emergency!!!"))
    }

    @Test
    fun testClassifiesQuestion() {
        assertEquals(IntentCategory.QUESTION, IntentClassifier.classify("Are you coming tonight?"))
        assertEquals(IntentCategory.QUESTION, IntentClassifier.classify("How are you"))
    }

    @Test
    fun testClassifiesTime() {
        assertEquals(IntentCategory.TIME, IntentClassifier.classify("Let's meet tomorrow morning"))
        assertEquals(IntentCategory.TIME, IntentClassifier.classify("See you at 8pm"))
    }

    @Test
    fun testClassifiesLocation() {
        assertEquals(IntentCategory.LOCATION, IntentClassifier.classify("Meet me at the station"))
    }

    @Test
    fun testClassifiesRequest() {
        assertEquals(IntentCategory.REQUEST, IntentClassifier.classify("Please send the file"))
        assertEquals(IntentCategory.QUESTION, IntentClassifier.classify("Could you help with this"))
    }

    @Test
    fun testClassifiesNegation() {
        assertEquals(IntentCategory.NEGATION, IntentClassifier.classify("No, I don't think so"))
    }

    @Test
    fun testClassifiesAffirmation() {
        assertEquals(IntentCategory.AFFIRMATION, IntentClassifier.classify("Yes, sounds good"))
    }

    @Test
    fun testClassifiesGreeting() {
        assertEquals(IntentCategory.GREETING, IntentClassifier.classify("Hello there"))
        assertEquals(IntentCategory.GREETING, IntentClassifier.classify("Good morning!"))
    }

    @Test
    fun testClassifiesFarewell() {
        assertEquals(IntentCategory.FAREWELL, IntentClassifier.classify("Bye, take care"))
    }

    @Test
    fun testClassifiesAcknowledgement() {
        assertEquals(IntentCategory.ACKNOWLEDGEMENT, IntentClassifier.classify("Got it, thanks"))
    }

    @Test
    fun testFallsBackToGeneral() {
        assertEquals(IntentCategory.GENERAL, IntentClassifier.classify("The quarterly report numbers look fine"))
    }

    @Test
    fun testIsDeterministic() {
        val input = "Meet me at the station tomorrow at 8pm, urgent!!"
        assertEquals(IntentClassifier.classify(input), IntentClassifier.classify(input))
    }

    @Test
    fun testNeverThrowsOnUnusualInput() {
        IntentClassifier.classify("😊".repeat(50))
        IntentClassifier.classify("你好，世界！")
    }
}
